pkg/queue/stats.go

```diff
@@ -29,9 +29,15 @@ func ReportStats(startedAt time.Time, reqCh chan network.ReqEvent, reportCh <-ch
 
 	for {
 		select {
-		case event := <-reqCh:
+		case event, ok := <-reqCh:
+			if !ok {
+				return
+			}
 			state.HandleEvent(event)
-		case now := <-reportCh:
+		case now, ok := <-reportCh:
+			if !ok {
```

Comment:
technically this case can never happen because ticker channels don't close - wdyt about just keeping the reqCh case?

Reply:
sure.